File: OficinaLuthier/API/Models/OrdemDeServico.cs
using System;

namespace API.Models;

public class OrdemDeServico
{
  public OrdemDeServico()
  {
    CriadoEm = DateTime.Now;
  }
  public int OrdemDeServicoId { get; set; }
  public string? DescricaoServico { get; set; }
  public string? Instrumento { get; set; }
   public string? Status { get; set; }
  public double ValorEstimado { get; set; }
  public Cliente? Cliente { get; set; }
  public int ClienteId { get; set; } 
  public Funcionario? Funcionario { get; set; }
  public int FuncionarioId { get; set; }
  public DateTime CriadoEm { get; set; }
}

File: front_luthieria/src/models/OrdemDeServico.ts
import { Funcionario } from './Funcionario';
import { Cliente } from "./Cliente";

export interface OrdemDeServico{
  ordemDeServicoId?: number,
  descricaoServico: string,
  instrumento: string,
  status: string,
  statusOptions?: StatusOrdemDeServico,
  valorEstimado: number,
  clienteId: number,
  cliente?: Cliente,
  funcionarioId: number,
  funcionario?: Funcionario,
  criadoEm?: string,
}

export type StatusOrdemDeServico = 'Pendente' | 'Em Andamento' | 'Finalizado';